### PR TITLE
Refactor Pretty Logic to frontend 

### LINKED ISSUE #1384: Refactor pretty logic from backend to frontend
## Description
Refactor the _pretty_name logic to frontend for maintaining API design consistency.

## Context
The context for this request is to support #1381 (Show unpretty input/output names in node details) and also adhere to the API design of not sending query parameters.

### PR DESCRIPTION
**Pull request by ravi-kumar-pilla**
## Description

Resolves #1384 

## Development notes

- Moved pretty and stripNamespace Logic from backend to frontend
- Removed full_name attribute from node schema. Every node should contain an attribute name.
- Replaced full_name attribute in class methods for creating node with dataset_name
- Renamed prettyName state variable to isPrettyName and also adjusted related functions and dispatch methods

## QA notes

- The metadata panel displays node name, inputs/outputs according to the pretty name flag set by the user in the settings modal
- Modified test cases, mock json responses to accommodate the refactor and variable name changes

## Checklist

- [x] Read the [contributing](/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added new entries to the `RELEASE.md` file
- [x] Added tests to cover my changes

<a href="https://gitpod.io/#https://github.com/kedro-org/kedro-viz/pull/1396"></a>

### CODE REVIEW

File: src/utils/index.js [hunk 1]
```diff
@@ -77,3 +77,43 @@ export const replaceMatches = (str, toReplace) => {
...
+  return str.replace(pattern, '');
+};
+
+/**
+ * Replaces any parts of a string that match the pattern with the target pattern and capitalizes each word in the string separated by a space
+ * @param {String} str The string to check
+ * @returns {String} The string with or without replaced values
+ */
+export const prettifyName = (str) => {
+  str = str.replace(/-/g, ' ').replace(/_/g, ' ').replace(/:/g, ': ').trim();
```
Comment (tynandebold) [inline src/utils/index.js:97]: I think we can assign this to a new variable here.

```suggestion
  const replacedString = str.replace(/-/g, ' ').replace(/_/g, ' ').replace(/:/g, ': ').trim();
```

File: src/utils/index.js [hunk 2]
```diff
@@ -77,3 +77,43 @@ export const replaceMatches = (str, toReplace) => {
...
+/**
+ * Prettifies name property of the nested object in a modularPipeline
+ * @param {Object} modularPipelines The object whose nested object property needs to be prettified
+ * @returns {Object} The object with or without prettified name inside the nested object
+ */
+export const prettifyModularPipelineNames = (modularPipelines) => {
+  for (const key in modularPipelines) {
+    if (modularPipelines.hasOwnProperty(key)) {
+      const modularPipeline = modularPipelines[key];
+      if (modularPipeline.hasOwnProperty('name')) {
```
Comment (tynandebold) [inline src/utils/index.js:110]: Tiny nit: spacing for legibility.

```suggestion
      const modularPipeline = modularPipelines[key];
      
      if (modularPipeline.hasOwnProperty('name')) {
```

**Review by tynandebold — APPROVED** [review]
Fantastic work here, Ravi! I tested this quite thoroughly against what's on our demo site and everything matched up. Great, great stuff 💥

I think the errors we're seeing on CircleCI aren't from what you've done and will be fixed with Rashida's PR here, https://github.com/kedro-org/kedro-viz/pull/1399, so I'm going to approve this now.

**Review by rashidakanchwala — APPROVED** [review]
LGTM, thanks!

**Review by Huongg — APPROVED** [review]
sorry for my late review. All looks good to me too 😄

**Pull request merged**